mapie/classification.py

```diff
@@ -943,15 +952,38 @@ def fit(
 
         estimator = check_estimator_classification(
             X,
-            y_enc,
+            y,
             cv,
             self.estimator
         )
         self.n_features_in_ = check_n_features_in(X, cv, estimator)
 
         n_samples = _num_samples(y)
-        self.n_classes_ = len(np.unique(y))
-        self.classes_ = np.unique(y)
+
+        if self.cv == "prefit":
+            self.classes_ = estimator.classes_
+            self.n_classes_ = len(np.unique(self.classes_))
+            if self.n_classes_ == len(np.unique(y)):
```

Comment:
Give a name to the `len(np.unique(y))`, you use it enough and a understandable variable name would give a better insight into what it is.

Reply:
done